### PR TITLE
test: create test add children

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Related Issues

#998

### Description and Example

Create the instrumentation test to 

### Checklist

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by carlossteinzup — DISMISSED** [review]

**Review by matheusribeirozup — DISMISSED** [review]

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/AddChildrenScreenSteps.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.hamcrest.Matchers.not
+import org.junit.Rule
+
+class AddChildrenScreenSteps {
+
+    @Rule
+    var activityTestRule = ActivityTestRule(MainActivity::class.java)
+
+    private var textFixedPosition: Int? = null
+    private var textAddedPosition: Int? = null
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/AddChildrenScreenSteps.kt:46]: are these variables being used?

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/elements/ScreenElements.kt [hunk 2]
```diff
@@ -85,3 +85,8 @@ const val SEND_BUTTON = "Enviar"
 const val OK_BUTTON = "OK"
 const val CANCEL_BUTTON = "Cancel"
 
+//AddChildren Screen
+const val ADD_CHILDREN_HEADER = "Add Children"
+const val TEXT_FIXED = "I'm the single text on screen"
+const val TEXT_ADDED = "New text added"
+const val CONTAINER_ID = "containerId"
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/elements/ScreenElements.kt:92]: What do you think about put this constant inside the add children test?

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/AddChildrenScreenSteps.kt (hunk 1 above)
Comment (viniciusguardieirozup) [reply android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/AddChildrenScreenSteps.kt]: No, now I remove this variables, thank you!

**Review by uziasferreirazup — DISMISSED** [review]

**Review by carlossteinzup — DISMISSED** [review]

**Comment by theffc** [comment]
It seems like this PR should not be named `feat:` since it's not adding some new feature for users, it instead adds an internal test. So, I would suggest we use some other prefix in these PRs that only change integration tests, maybe `test: ` would be good

**Comment by theffc** [comment]
I miss clicked and closed this PR, I'm sorry 😅

File: iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                return
+            }
+            screen.clickOnButton(button: element)
+        }
+        
+        Then("^A Text need to be added after the already exist one$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.TEXT_ADDED.element.exists)
+        }
+        
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift:46]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d22cba59d79879d25025e40e0c3e238ccf52e76f/iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift#L46"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>AddChildrenSteps.swift:46</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        Then("^A Text need to be added after the already exist one$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.TEXT_ADDED.element.exists)
+        }
+        
+        
+        Then("^A Text need to be added before the already exist one$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.TEXT_FIXED.element.exists)
+        }
+        
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift:51]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d22cba59d79879d25025e40e0c3e238ccf52e76f/iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift#L51"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>AddChildrenSteps.swift:51</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            XCTAssertTrue(ScreenElements.TEXT_FIXED.element.exists)
+        }
+        
+        
+        Then("^A Text need to replace the already exist one$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.TEXT_ADDED.element.exists)
+            XCTAssertFalse(ScreenElements.TEXT_FIXED.element.exists)
+        }
+        
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift:57]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d22cba59d79879d25025e40e0c3e238ccf52e76f/iOS/Tests/AutomatedTestsUITests/Steps/AddChildrenSteps.swift#L57"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>AddChildrenSteps.swift:57</code></td>
    </tr>
  </tbody>
</table>

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/AddChildrenScreenBuilder.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    )
+
+    private fun createButton(text : String, actions : List<Action>) = Button(text = text, onPress = actions)
+
+    private fun createActions(componentId : String, mode : Mode? = null) = listOf<Action>(AddChildren(
+        componentId = componentId,
+        mode = mode,
+        value = listOf(Text(text = "New text added"))
+    ))
+}
```
Comment (victorrabelozup) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/AddChildrenScreenBuilder.kt:58]: Why this class is duplicated?

File: iOS/Tests/AutomatedTestsUITests/Features/AddChildren.feature [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        When I click on button "PREPEND COMPONENT NULL"
+        Then Nothing should happen
+
+    Scenario: Replace - AddChildren replace the children container to a Text
+        When I click on button "APPEND COMPONENT NULL"
+        Then Nothing should happen
+
+    Scenario: Replace - AddChildren replace the children container to a Text
+        When I click on button "REPLACE COMPONENT NULL"
+        Then Nothing should happen
```
Comment (victorrabelozup) [inline iOS/Tests/AutomatedTestsUITests/Features/AddChildren.feature:52]: why is this class duplicated?

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/AddChildrenScreenBuilder.kt (hunk 6 above)
Comment (viniciusguardieirozup) [reply backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/AddChildrenScreenBuilder.kt:58]: I added an example of this use on sample, and for the automated tests, we need to have an screen on package AutomatedTests and this screen can be the same, we don't have a common for sample and automated to remove the duplicated codes

**Review by victorrabelozup — DISMISSED** [review]

**Review by uziasferreirazup — DISMISSED** [review]

**Review by caioortu — APPROVED** [review]

**Review by gabcoelho — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**